### ISSUE TITLE
Issue with implementation of MMSU Stop and Status Scripts in Cron job

### ISSUE THREAD

**Issue by mayankcp** [sassoftware/virk#96]
In below scenarios, MMSU Stop and Status scripts do not pick up the running Viya Services when you schedule in Cronjob.
It seems to me "enable_stray_cleanup: true  #stray processes cleanup" does not work with Cronjob and we are not grepping status of Viya services at Operating System level e.g. ps -ef | grep viya

**Problem:**

SAS Viya 3.4 - In Multi-machine deployment of Viya (in AWS Quick Start) - All Viya services were working as expected. Since implementing the VIRK utility playbook to stop and start Viya services using **CRONJOB**; after rebooting the Viya servers I was getting below error message in logs but when I ran playbook manually all Viya Services started correctly:

_PLAY [rabbitmq] ****************************************************************

TASK [Start SAS Message Broker (RabbitMQ)] *************************************
fatal: [stateful]: FAILED! => {"changed": false, "msg": "Unable to start service sas-viya-rabbitmq-server-default: Job for sas-viya-rabbitmq-server-default.service failed because a timeout was exceeded. See \"systemctl status sas-viya-rabbitmq-server-default.service\" and \"journalctl -xe\" for details.\n"}

NO MORE HOSTS LEFT *************************************************************
        to retry, use: --limit @/home/ec2-user/sas_viya_playbook/virk/playbooks/service-management/viya-services-start.retry_

**Error in Playbook logs:**

![error in playbook logs](https://user-images.githubusercontent.com/30283214/50742727-33b3e280-1263-11e9-9240-0a7cb5f9c5aa.jpg)

**Troubleshooting:**

Run “chkconfig –list” to confirm the runtime status of Viya services. In my case, Runtime value of “sas-viya-vault-default” service was set to “on” for 2-5. Some reason “viya-services-disable.yml” playbook failed to update all Viya services.

sas-viya-vault-default  0:off   1:off   2:on    3:on    4:on    5:on    6:off

Use “ps -ef | grep viya” in all Viya servers to find status of Viya services

In my case RabbitMQ Services were still running:
![rabbimq services](https://user-images.githubusercontent.com/30283214/50742751-55ad6500-1263-11e9-9004-b1d0554c4557.jpg)

**Resolution:**

•	In Ansbile server viya-services-vars.yml
Set below parameter value to “true”
enable_stray_cleanup: true  #stray processes cleanup
•	Update below parameter values in “viya-services-disable.yml” playbook
name: sas-viya-vault-default
enabled: no
•	Run above playbook
Run “chkconfig –list” to confirm the runtime status of “sas-viya-vault-default” service. All runtime values must be “off”
sas-viya-vault-default  0:off   1:off   2:off    3:off    4:off    5:off    6:off
•	Schedule the cron job to start Viya Services in Ansible server.
•	Stop all Viya services and Shutdown all Viya Servers.
•	Start all Viya Servers
•	Check all Viya services status – they should be all up and running

**Example for Cronjob:**

**Script (viya-services-stop.sh):**
#!/bin/bash
/usr/local/bin/ansible-playbook -i /home/ec2-user/sas_viya_playbook/inventory.ini  /home/ec2-user/sas_viya_playbook/virk/playbooks/service-management/viya-services-stop.yml

**Crontab:**
#### Stop all Viya Services at 1630hrs Mon-Fri #######
27 14 * * 1-5 /home/ec2-user/sas_viya_playbook/virk/playbooks/service-management/viya-services-stop.sh > /var/log/sas/viya-services-stop-$(date +\%Y\%m\%d\%H\%M\%S).log

**How to reproduce this problem please see attached file.**
[MMU_Script_Stop_Viya34_Reproduce_Problem.docx](https://github.com/sassoftware/virk/files/2730982/MMU_Script_Stop_Viya34_Reproduce_Problem.docx)

**Comment by cuddlehub** [comment]
For out of box, sas-viya-vault-default wasn't delievered  to be enabled at boot time. It appears that your environment has been modified. viya-services-disable.yml is used to disable sas-viya-all-services which is the only service enabled at boot time during install.

**Comment by mayankcp** [comment]
> For out of box, sas-viya-vault-default wasn't delievered to be enabled at boot time. It appears that your environment has been modified. viya-services-disable.yml is used to disable sas-viya-all-services which is the only service enabled at boot time during install.

Thanks. Yes, I'm aware of that. We used Viya AWS Quick Start template to deploy Viya in customer's AWS account. Then we implemented the AWS Instance Scheduler to stop and start Viya servers and as part of that we used viya-services-disable.yml playbook to disable all Viya services after bootup but some reason viya-services-disable.yml playbook failed to disable the sas-viya-vault-default (Re-ran the few times after we noticed the issue but failed to update the sas-viya-vault-default service although, no error reported) so to fix it ended up creating a new playbook "viya-services-disable.yml" with "sas-viya-vault-default" value.

Regardless, when we run Playbook "viya-services-stop playbook work" manually, it works but not through cronjob. After enabling the "enable_stray_cleanup: true" it should pick up any running Viya services regardless though Cronjob or manually. It might be a problem with sas-viya-all-services stop which shipped as part of Viya Install package.
